vllm/worker/enc_dec_model_runner.py

```diff
@@ -0,0 +1,719 @@
+        # vLLM blocker manager.
+        # To exercise the worst scenario for GPU memory consumption,
+        # the number of seqs (batch_size) is chosen to maximize the number
+        # of images processed.
+        model_config = self.model_config
+
+        if supports_vision(self.model):
+            max_mm_tokens = MULTIMODAL_REGISTRY \
+                .get_max_multimodal_tokens(model_config)
+            max_num_seqs_orig = max_num_seqs
+            max_num_seqs = min(max_num_seqs,
+                               max_num_batched_tokens // max_mm_tokens)
+            if max_num_seqs < 1:
+                expr = (f"min({max_num_seqs_orig}, "
+                        f"{max_num_batched_tokens} // {max_mm_tokens})")
+                logger.warning(
+                    "Computed max_num_seqs (%s) to be less than 1. "
+                    "Setting it to the minimum value of 1.", expr)
+                max_num_seqs = 1
+
+        batch_size = 0
+        for group_id in range(max_num_seqs):
+            seq_len = (max_num_batched_tokens // max_num_seqs +
+                       (group_id < max_num_batched_tokens % max_num_seqs))
+            batch_size += seq_len
+
+            seq_data, dummy_multi_modal_data = INPUT_REGISTRY \
+                .dummy_data_for_profiling(model_config, seq_len)
+
+            # Having more tokens is over-conservative but otherwise fine
+            assert len(seq_data.prompt_token_ids) >= seq_len, (
+                f"Expected at least {seq_len} dummy tokens for profiling, "
+                f"but got: {len(seq_data.prompt_token_ids)}")
+
+            seq = SequenceGroupMetadata(
+                request_id=str(group_id),
+                is_prompt=True,
+                seq_data={group_id: seq_data},
+                sampling_params=sampling_params,
+                block_tables=None,
+                lora_request=dummy_lora_requests_per_seq[group_id]
+                if dummy_lora_requests_per_seq else None,
+                encoder_seq_data=seq_data,
+                cross_block_table=None,
+                multi_modal_data=dummy_multi_modal_data,
+            )
+            seqs.append(seq)
+
+        # Run the model with the dummy inputs.
+        num_layers = self.model_config.get_num_layers(self.parallel_config)
+        kv_caches = [None] * num_layers
+        finished_requests_ids = [seq.request_id for seq in seqs]
+        model_input = self.prepare_model_input(
+            seqs, finished_requests_ids=finished_requests_ids)
+        intermediate_tensors = None
+        if not get_pp_group().is_first_rank:
+            intermediate_tensors = self.model.make_empty_intermediate_tensors(
+                batch_size=batch_size,
+                dtype=self.model_config.dtype,
+                device=self.device)
+        self.execute_model(model_input, kv_caches, intermediate_tensors)
+        torch.cuda.synchronize()
+        return
+
+    def _prepare_encoder_model_input_tensors(
```

Comment:
I tried this briefly. The thing is, that no matter what the model_input data structure has to be updated with the encoder prompt attributes. Right now that happens inside of `_prepare_encoder_model_input_tensors()`, presumably in the implementation you are proposing the code to update the model_input data structure would have to be handled by the code which invoked `_prepare_encoder_model_input_tensors()` instead. I don't feel that this decision reduces the complexity of `_prepare_encoder_model_input_tensors()` significantly, and in exchange it adds complexity to every instance where `_prepare_encoder_model_input_tensors()` is called. 

Also, your code snippet implies that the implementation can be broken cleanly into an `is_prompt` branch and a `not is_prompt` branch:


```
     if is_prompt:
         # Compute the slot mappings for the cross attention block table, 
         # the encoder_input_tokens and  encoder_input_positions
     else:
         # Only the cross attention block table info is needed
```

I tried writing the implementation in this way and concluded that, while in principle you can group together all of the `is_prompt`-scenario code and all of the `not is_prompt`-scenario code into separate branches, in practice this does not significantly reduce complexity - and perhaps even increases complexity - because there is a significant amount of code that is common to both scenarios.

So in summary, I decided not to go forward with the implementation you proposed above.

However, I *did* make a number of changes to `_prepare_encoder_model_input_tensors()` which significantly reduce complexity, mostly by removing all unsupported feature code (prefix caching, sliding window, chunked prefill, as-of-yet-unsupported kernels, etc.)